pkg/server/server.go

```diff
@@ -84,10 +93,15 @@ func (host *yamlLanguageHost) loadTemplate(directory string, compilerEnv []strin
 	if diags.HasErrors() {
 		return nil, diags, nil
 	}
-	host.template = template
-	host.diags = diags
 
-	return host.template, diags, nil
+	if host.compiler != "" {
```

Comment:
Might be worth either adding a `shouldCache` method or even just a `cache` method that internalise this check/provide a single place we can document it.